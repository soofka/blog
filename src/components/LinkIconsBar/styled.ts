import styledComponents from 'styled-components';

export const StyledLinkIconsBar = styledComponents.ul`
  margin: ${props => props.theme.marginM};
  padding: 0;
  text-align: center;

  li {
    display: inline-block;
    margin: ${props => `${props.theme.marginM} ${props.theme.marginXS}`};
    padding: 0;
    font-size: ${props => props.theme.fontSizeXXL};
    line-height: ${props => props.theme.fontSizeXXL};

    svg {
      width: 40px;
      height: 40px;
      opacity: 0.7;

      &:hover,
      &:active {
        opacity: 1;
      }
    }
  }

  @media (max-width: ${props => props.theme.mobileBreakPointWidth}) {
    float: right;
    margin: ${props => props.theme.marginS} 0;

    li {
      margin: 0 ${props => props.theme.marginXS};
      padding: 0;
      font-size: ${props => props.theme.fontSizeL};
      line-height: ${props => props.theme.fontSizeL};
    }
  }
`;

export default StyledLinkIconsBar;
