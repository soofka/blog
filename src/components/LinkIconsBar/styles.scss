@import 'components/Blog/variables.scss';

.link-icons-bar {
  margin: $marginM;
  padding: 0;
  text-align: center;

  li {
    display: inline-block;
    margin: $marginM $marginXS;
    padding: 0;
    font-size: $fontSizeXXL;
    line-height: $fontSizeXXL;

    svg {
      width: 40px;
      height: 40px;
      opacity: 0.7;

      &:hover,
      &:active {
        opacity: 1;
      }
    }
  }
}

@media (max-width: $mobileBreakPointWidth) {
  .link-icons-bar {
    float: right;
    margin: $marginS 0;

    li {
      margin: 0 $marginXS;
      padding: 0;
      font-size: $fontSizeL;
      line-height: $fontSizeL;
    }
  }
}
