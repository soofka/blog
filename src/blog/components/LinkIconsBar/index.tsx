import * as React from 'react';

import LinkedInIcon from './icons/LinkedInIcon';
import TwitterIcon from './icons/TwitterIcon';
import GitHubIcon from './icons/GitHubIcon';
import InstagramIcon from './icons/InstagramIcon';
import LastFmIcon from './icons/LastFmIcon';

import { StyledLinkIconsBar } from './styled';

export interface SvgIconPropsInterface {
  fill?: string;
}

export const LinkIconsBar = (): JSX.Element => {
  const color = '#c3c3c3';
  return (
    <StyledLinkIconsBar>
      <li>
        <LinkedInIcon fill={color} />
      </li>
      <li>
        <TwitterIcon fill={color} />
      </li>
      <li>
        <GitHubIcon fill={color} />
      </li>
      <li>
        <InstagramIcon fill={color} />
      </li>
      <li>
        <LastFmIcon fill={color} />
      </li>
    </StyledLinkIconsBar>
  );
};

export default LinkIconsBar;
