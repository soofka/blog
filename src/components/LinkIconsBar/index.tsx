import * as React from 'react';

import LinkedInIcon from './icons/LinkedInIcon';
import TwitterIcon from './icons/TwitterIcon';
import GitHubIcon from './icons/GitHubIcon';
import InstagramIcon from './icons/InstagramIcon';
import LastFmIcon from './icons/LastFmIcon';

import './styles.scss';

export interface SvgIconPropsInterface {
  fill?: string;
}

export const LinkIconsBar = (): JSX.Element => {
  return (
    <ul className="link-icons-bar">
      <li>
        <LinkedInIcon fill="#c3c3c3" />
      </li>
      <li>
        <TwitterIcon fill="#c3c3c3" />
      </li>
      <li>
        <GitHubIcon fill="#c3c3c3" />
      </li>
      <li>
        <InstagramIcon fill="#c3c3c3" />
      </li>
      <li>
        <LastFmIcon fill="#c3c3c3" />
      </li>
    </ul>
  );
};

export default LinkIconsBar;
