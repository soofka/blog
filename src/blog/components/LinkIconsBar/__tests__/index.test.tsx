import * as React from 'react';
import { shallow } from 'enzyme';

import LinkIconsBar from 'components/LinkIconsBar';

describe('<LinkIconsBar/>', () => {

  it('renders correctly', () => {
    expect(shallow(<LinkIconsBar/>)).toMatchSnapshot();
  });

});
