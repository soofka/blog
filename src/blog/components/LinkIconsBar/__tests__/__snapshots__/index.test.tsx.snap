// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkIconsBar/> renders correctly 1`] = `
<styled.ul>
  <li>
    <Component
      fill="#c3c3c3"
    />
  </li>
  <li>
    <Component
      fill="#c3c3c3"
    />
  </li>
  <li>
    <Component
      fill="#c3c3c3"
    />
  </li>
  <li>
    <Component
      fill="#c3c3c3"
    />
  </li>
  <li>
    <Component
      fill="#c3c3c3"
    />
  </li>
</styled.ul>
`;
